.travis.yml

```diff
@@ -81,6 +84,29 @@ regressiontest: &regressiontest
     - coverage xml --ignore-errors
     - bash <(curl -s https://codecov.io/bash)
 
+bash: &bash
+  stage: functional-test-bash
+  python:
+    - "3.7"
+  before_install:
+    - docker build -t bash:test dockerfiles/bash/
+    # does not have a supervisor or daemon process entrypoint, so needs both -t and -d
+    - docker run --name bash -v $(pwd -P):/root/cylc-flow --rm -t -d bash:test /bin/bash
+    - docker ps -a
+  install:
+    - docker exec bash python3.7 -m pip install six==1.12
+    - docker exec -w /root/cylc-flow bash python3.7 -m pip install -e .[all]
+  script:
+    # we do not run all the tests, as we have too many bash versions to test!
+    - docker exec bash update-alternatives --set bash /bash/bash-4.1
+    - docker exec -e CYLC_TEST_RUN_GENERIC=true -w /root/cylc-flow bash ./etc/bin/run-functional-tests.sh -v tests/broadcast/00-simple.t
+    - docker exec bash update-alternatives --set bash /bash/bash-4.2
+    - docker exec -e CYLC_TEST_RUN_GENERIC=true -w /root/cylc-flow bash ./etc/bin/run-functional-tests.sh -v tests/broadcast/00-simple.t
+    - docker exec bash update-alternatives --set bash /bash/bash-4.3
+    - docker exec -e CYLC_TEST_RUN_GENERIC=true -w /root/cylc-flow bash ./etc/bin/run-functional-tests.sh -v tests/broadcast/00-simple.t
+    - docker exec bash update-alternatives --set bash /bash/bash-4.4
+    - docker exec -e CYLC_TEST_RUN_GENERIC=true -w /root/cylc-flow bash ./etc/bin/run-functional-tests.sh -v tests/broadcast/00-simple.t
```

Comment:
There's probably a much better way to run these steps, maybe a single shell taking parameters... but I didn't spend much time thinking about it as it's more of a PoC